### PR TITLE
ticket:3488 added new API functions for vendorannotation

### PR DESCRIPTION
**Pull request by arun3688**

### CODE REVIEW

File: Compiler/Script/CevalScriptBackend.mo [hunk 1]
```diff
@@ -1710,6 +1710,24 @@ algorithm
       then
         (cache,Values.TUPLE({Values.REAL(startTime),Values.REAL(stopTime),Values.REAL(tolerance),Values.INTEGER(numberOfIntervals),Values.REAL(interval)}),st);
 
+    case (cache,_,"getAnnotationNamedModifiers",{Values.CODE(Absyn.C_TYPENAME(classpath)),Values.STRING(annotationname)},st as GlobalScript.SYMBOLTABLE(ast=p),_)
+      equation
+          Absyn.CLASS(cname,_,_,_,_,cdef,_) =Interactive.getPathedClassInProgram(classpath,p);
+          annlst= getannotationlist(cdef);
+          modifiernamelst=getelementargsmodifiers(annlst,annotationname);
```
Comment (sjoelund) [inline Compiler/Script/CevalScriptBackend.mo]: Why all lower case function names? Existing functions are mostly camelCase

File: Compiler/Script/CevalScriptBackend.mo [hunk 2]
```diff
@@ -6734,5 +6752,188 @@ algorithm
...
+    local
+      list<Absyn.ElementArg> elt,eltarglst;
+      Option<Absyn.Modification> modification;
+      String name,name1,name2,s;
+      String modfiername_value;
+      Absyn.ElementArg eltarg;
+
+  case (Absyn.MODIFICATION(_,_, Absyn.IDENT(name),modification,_)::eltarglst,name1,name2)
+    equation
+       true = stringEq(name, name1);
```
Comment (sjoelund) [inline Compiler/Script/CevalScriptBackend.mo]: Do not use matchcontinue for these things. Use match + guard

File: Compiler/Script/CevalScriptBackend.mo [hunk 3]
```diff
@@ -6734,5 +6752,188 @@ algorithm
...
+
+  case({},name1,name2) then "";
+  end matchcontinue;
+ end getelementargsmodifiersValue;
+
+function getelementargslist
+  "@author arun Helper function which gives list of elementargs from modification"
+  input Option<Absyn.Modification> inmod;
+  output list<Absyn.ElementArg> outargs;
+  algorithm
```
Comment (sjoelund) [inline Compiler/Script/CevalScriptBackend.mo]: Odd indentation in this function

File: Compiler/Script/CevalScriptBackend.mo [hunk 4]
```diff
@@ -6734,5 +6752,188 @@ algorithm
...
+  outargs:=matchcontinue(inmod)
+   local
+   list<Absyn.ElementArg> elementargs;
+   Absyn.EqMod eqmod;
+   case(SOME(Absyn.CLASSMOD(elementargs,eqmod)))
+     equation
```
Comment (sjoelund) [inline Compiler/Script/CevalScriptBackend.mo]: Unnecessary equation keyword

File: Compiler/Script/CevalScriptBackend.mo [hunk 5]
```diff
@@ -6734,5 +6752,188 @@ algorithm
...
+     end matchcontinue;
+end getelementargslist;
+
+
+function getmodifiersnamelist
+"@author arun Function which retrives vendor annotation modifiers name list"
+  input list<Absyn.ElementArg> eltArgs;
+  output list<String> strs;
+algorithm
+  (strs) := matchcontinue (eltArgs)
```
Comment (sjoelund) [inline Compiler/Script/CevalScriptBackend.mo]: Do not use matchcontinue unless necessary

File: Compiler/Script/CevalScriptBackend.mo [hunk 6]
```diff
@@ -6734,5 +6752,188 @@ algorithm
...
+   Absyn.EqMod eqmod;
+   case(SOME(Absyn.CLASSMOD(elementargs,eqmod)))
+     equation
+     then
+     elementargs;
+     end matchcontinue;
+end getelementargslist;
+
+
+function getmodifiersnamelist
```
Comment (sjoelund) [inline Compiler/Script/CevalScriptBackend.mo]: This is a mapping function. Use a list comprehension to improve readability

File: Compiler/Script/CevalScriptBackend.mo [hunk 7]
```diff
@@ -6734,5 +6752,188 @@ algorithm
...
+    local
+
+      list<Absyn.ElementArg> elt;
+      Absyn.Exp e;
+      list<Absyn.ElementArg> eltarglst;
+      String name,name1,s,modfiername_value;
+
+   case (Absyn.MODIFICATION(_,_, Absyn.IDENT(name), SOME(Absyn.CLASSMOD(elt, Absyn.EQMOD(e))))::eltarglst,name1)
+    equation
+       true = stringEq(name, name1);
```
Comment (sjoelund) [inline Compiler/Script/CevalScriptBackend.mo]: Use guards

**Comment by sjoelund** [comment]
Which pull request has the corresponding test cases?

**Comment by arun3688** [comment]
Fixed all the comments, the pull request, #324 contains the test case for the API.

File: Compiler/Script/CevalScriptBackend.mo [hunk 8]
```diff
@@ -6734,5 +6751,166 @@ algorithm
...
+algorithm
+  strs := list(match mod case (Absyn.MODIFICATION(_,_, Absyn.IDENT(name),_)) then name; end match for mod in eltArgs);
+end getModifiersNameList;
+
+function getModifierNamedValue
+" @author arun Function which retrives vendor annotation modifiername value"
+  input list<Absyn.ElementArg> eltArgs;
+  input String instring;
+  output String strs;
+algorithm
```
Comment (sjoelund) [inline Compiler/Script/CevalScriptBackend.mo]: This can still be a list comprehension

File: Compiler/Script/CevalScriptBackend.mo [hunk 9]
```diff
@@ -6734,5 +6751,176 @@ algorithm
...
+    outstring:=matchcontinue(inargs,instring)
+    local
+    list<Absyn.ElementArg> elt,eltarglst;
+    Option<Absyn.Modification> modification;
+    String name,name1, s;
+    list<String> modfiernamelist;
+    Absyn.ElementArg eltarg;
+  case (Absyn.MODIFICATION(_,_, Absyn.IDENT(name),modification,_)::eltarglst,name1)
+    equation
```
Comment (sjoelund) [inline Compiler/Script/CevalScriptBackend.mo]: This could be a guard without matchcontinue

File: Compiler/Script/CevalScriptBackend.mo [hunk 10]
```diff
@@ -6734,5 +6751,176 @@ algorithm
...
 end getComponentitemsName;
 
+
+// new functions added for getting vendorannotation modifiers names and their values
+function getAnnotationList
+   "@author arun Helper function which returns the list of annotation items as elementargs "
+   input Absyn.ClassDef inclassdef;
+   output list<Absyn.ElementArg> anninfo;
+algorithm
+   anninfo:=matchcontinue(inclassdef)
```
Comment (sjoelund) [inline Compiler/Script/CevalScriptBackend.mo]: This is a match expression

File: Compiler/Script/CevalScriptBackend.mo [hunk 11]
```diff
@@ -6734,5 +6751,176 @@ algorithm
...
+algorithm
+   outstring:=matchcontinue(inargs,instring,instring1)
+    local
+      list<Absyn.ElementArg> elt,eltarglst;
+      Option<Absyn.Modification> modification;
+      String name,name1,name2,s;
+      String modfiername_value;
+      Absyn.ElementArg eltarg;
+
+    case (Absyn.MODIFICATION(_,_, Absyn.IDENT(name),modification,_)::eltarglst,name1,name2)
```
Comment (sjoelund) [inline Compiler/Script/CevalScriptBackend.mo]: This is a match expression

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2016-05-25_13-11-12](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/2796/console).

**Comment by casella** [comment]
Does this resolve ticket:3488?

**Comment by adeas31** [comment]
No. This is just one step towards it.

**Comment by adrpo** [comment]
@casella you can see what is left to do in: https://trac.openmodelica.org/OpenModelica/ticket/3934

**Pull request closed without merging**